tools/defaults2.bzl

```diff
@@ -0,0 +1,120 @@
+load("@aspect_bazel_lib//lib:copy_to_bin.bzl", "copy_to_bin")
+load("@aspect_bazel_lib//lib:expand_template.bzl", "expand_template")
+load("@aspect_bazel_lib//lib:jq.bzl", "jq")
+load("@aspect_bazel_lib//lib:utils.bzl", "to_label")
+load("@aspect_rules_jasmine//jasmine:defs.bzl", _jasmine_test = "jasmine_test")
+load("@aspect_rules_js//npm:defs.bzl", _npm_package = "npm_package")
+load("@rules_pkg//:pkg.bzl", "pkg_tar")
+load("//tools:interop.bzl", _ts_project = "ts_project")
+load("//tools:link_package_json_to_tarballs.bzl", "link_package_json_to_tarballs")
+load("//tools:snapshot_repo_filter.bzl", "SNAPSHOT_REPO_JQ_FILTER")
+load("//tools:substitutions.bzl", "NO_STAMP_PACKAGE_SUBSTITUTIONS", "get_npm_package_substitutions_for_rjs")
+
+def jasmine_test(**kwargs):
+    _jasmine_test(
+        node_modules = "//:root_modules",
+        **kwargs
+    )
+
+def ts_project(**kwargs):
+    _ts_project(**kwargs)
+
+def npm_package(
```

Comment:
This is complicated enough that it might warrant going into its own bzl file? A lot happens here for the npm package.